### PR TITLE
Add port_connectivity datapoint for hosts

### PR DESCRIPTION
**Pull request by sdodsley**
Here we go - first-ever stab at patching this code.
Probably very wrong...

### CODE REVIEW

**Comment by chrroberts-pure** [comment]
initial review looks good. I'll give it a test and give final approval tomorrow first thing!

**Comment by chrroberts-pure** [comment]
please also make sure to update specification/metrics/purefa-metrics.md

**Comment by sdodsley** [comment]
@chrroberts-pure Updated with available options

**Comment by chrroberts-pure** [comment]
I attempted to run as is on a FlashArray running 6.5.0 and received the following results. 

````
purefa_host_connections_info{connectivity="",host="bob1",hostgroup="",volume="GP1"} 1
purefa_host_connections_info{connectivity="",host="bob2",hostgroup="",volume="GP1"} 1
````

I am only seeing `null` values for `connectivity` on host that are connected.

I am happy to help debug this with/for you - or if you'd like another shot let me know. Please refer to README.md on how to locally build and run the exporter.

**Comment by chrroberts-pure** [comment]
from my quick research - it looks like `port_connectivity` come from PurityAPI URI `/hosts`, and not from `/connections` as it's currently expecting by calling `GetConnections`

I would consider making an entirely new metric `purefa_host_connectivity_info` with `status` and `details` as Attribute 
 Keys.

**Comment by chrroberts-pure** [comment]
I moved the new metric to use `GetHosts`, and tested. Docs have been updated as well. I would like a review from @genegr and @james-laing asap to get this out with 1.0.10.

Example new metic output
```
# HELP purefa_host_connectivity_info FlashArray host connectivity information
# TYPE purefa_host_connectivity_info gauge
purefa_host_connectivity_info{details="N/A",host="host1",status="unused"} 1
purefa_host_connectivity_info{details="None",host="host2",status="critical"} 1
purefa_host_connectivity_info{details="Redundant",host="host3",status="healthy"} 1
```

File: internal/openmetrics-exporter/host_connections_collector_test.go
Comment (chrroberts-pure) [inline internal/openmetrics-exporter/host_connections_collector_test.go:1]: VSGo like to really format things correctly if they've been opened. Happy to remove this file.

File: internal/rest-client/connections.go
Comment (chrroberts-pure) [inline internal/rest-client/connections.go:1]: VSGo like to really format things correctly if they've been opened. Happy to remove this file.

**Review by chrroberts-pure — APPROVED** [review]
LGTM, and tested on a FlashArray running 6.5.0

**Comment by james-laing** [comment]
Functionality tested ok. However the binary, after some indeterminate amount of time fails with error
`fatal error: concurrent map writes`

**Comment by james-laing** [comment]
Raised https://github.com/PureStorage-OpenConnect/pure-fa-openmetrics-exporter/issues/85 for `fatal error: concurrent map writes` error not related to this PR.
Tested functionality and confirm it works.

**Pull request merged**